packages/common/src/hooks/useFeatureFlag.ts

```diff
@@ -146,25 +146,25 @@ export const useFeatureFlag = (
     [overrideKey, localStorage]
   )
 
-  const [isLocallyEnabled, setIsLocallyOverriden] = useState<Maybe<boolean>>()
+  const [isLocallyEnabled, setIsLocallyOverriden] =
+    useState<Maybe<Nullable<boolean>>>()
 
   useEffectOnce(() => {
     const getOverride = async () => {
       const override = await localStorage.getItem(overrideKey)
       if (override === 'enabled') {
         setIsLocallyOverriden(true)
-      }
-      if (override === 'disabled') {
+      } else if (override === 'disabled') {
         setIsLocallyOverriden(false)
+      } else {
+        setIsLocallyOverriden(null)
       }
-
-      return undefined
     }
     getOverride()
   })
 
   return {
-    isLoaded: configLoaded,
+    isLoaded: configLoaded && isLocallyEnabled !== undefined,
```

Comment:
Interesting, I like this! Maybe a comment would be good to clarify the null vs undefined state

Reply:
I'm generally in favor of `null` being an explicitly set empty value (vs. `undefined` meaning our code never touched it and it's empty).

I do think it's a little cryptic to derive "has localStorage been read" from the value of a variable we attempted to read out of local storage. An explicit boolean indicating that we finished fetching the item from localStorage might be better.

```ts
const [isLocallyOverridden, setIsLocallyOverridden] = useState<Maybe<boolean>>()
const [hasFetchedLocalStorage, setHasFetchedLocalStorage] = useState(false)

useEffectOnce(() => {
  const getOverride = async () => {
    const override = await localStorage.getItem(overrideKey)
    if (override === 'enabled') {
      setIsLocallyOverridden(true)
    } else if (override === 'disabled' { 
      setIsLocallyOverridden(false)
    }
    setHasFetchedLocalStorage(true)
  }
})

return {
  isLoaded: configLoaded && hasFetchedLocalStorage
  isEnabled: isLocallyOverridden ?? isEnabled
}
```


Also, wanted to point out that the variable name and the setter don't match and by convention they probably should to avoid confusion.